src/form/Form.js

```diff
@@ -78,6 +79,19 @@
         if (this.validate()) {
             isValid = this.checkValidation();
         }
+        if (!this.submitEmpty()) {
+            var hasValue = false;
+            for (var key in this.values()) {
+                var val = this.values()[key];
+                if (val !== undefined && val !== null && val.trim() !== "") {
+                    hasValue = true;
+                    break;
+                }
+            }  
+            if (!hasValue) {
+                return;
+            }
+        }
```

Comment:
Will need to provide some feedback to the user - ideally the submit button would be disabled until some data was entered?